### PR TITLE
Update design of Settings Preference page

### LINKED ISSUE #1142: Settings Preference page out of spec
### Dev Home version

0.137.150.0

### Windows build number

10.0.22621.1848

### Other software

OS Build Version: 10.0.22621.1848.amd64fre.ni_release.220506-1250
.NET Version: .NET 6.0.16

### Steps to reproduce the bug

Open Dev home > Click settings icon in lower left side nav > click Preferences. 

### Expected result

Currently, the preferences setting page has the theme selection via radio buttons when it should be shown as a drop down selector. Additionally, the 'Start Dev Home when I sign into Windows' and 'Hide get started content' features are missing. They are both controlled vis an on/off toggle 

### Actual result

Preference Settings page is missing two features controlled by on/off toggle switches. and the Theme selector is shown as radio buttons when it should be a dropdown list. 

### Included System Information

CPU: 11th Gen Intel(R) Core(TM) i7-1185G7 @ 3.00GHz
Physical Memory: 31.85GB (11.45GB free)
Processor Architecture: x64

### Included Extensions Information

Extensions: 
Microsoft.Windows.DevHome.GitHub_1.0.0.0_x64__8wekyb3d8bbwe
Microsoft.Windows.DevHomeGitHubExtension_0.137.141.0_x64__8wekyb3d8bbwe
Microsoft.Windows.DevHome_0.137.150.0_x64__8wekyb3d8bbwe

### PR DESCRIPTION
**Pull request by krschau**
## Summary of the pull request
The Preferences page in Settings currently looks like this:
image

But should look like this:
image

- Use a SettingsCard instead of a SettingsExpander
- Use a ComboBox rather than radio buttons
- Two-way bind to the index and use an EventTriggerBehavior to kick off the actual theme change
- Change EnumToBooleanConverter to EnumToIntConverter since now the binding is to the enum

I ordered the ComboBoxItems to have the same indexes as the ElementTheme enum for simplicities sake. If we wanted to change the order if the future, we'd need to add more logic to the converter.

## References and relevant issues
#1142

## PR checklist
- [ ] Closes #1142
- [ ] Tests added/passed
- [ ] Documentation updated

### CODE REVIEW

File: common/Helpers/EnumToIntConverter.cs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+using System;
+using Microsoft.UI.Xaml;
+using Microsoft.UI.Xaml.Data;
+
+namespace DevHome.Common.Helpers;
+
+public class EnumToIntConverter : IValueConverter
+{
+    public EnumToIntConverter()
```
Comment (dhoehna) [inline common/Helpers/EnumToIntConverter.cs:12]: Why the zero argument Constructor that does nothing?

File: common/Helpers/EnumToIntConverter.cs [hunk 2]
```diff
@@ -0,0 +1,43 @@
+﻿// Copyright (c) Microsoft Corporation and Contributors
+// Licensed under the MIT license.
+
+using System;
+using Microsoft.UI.Xaml;
+using Microsoft.UI.Xaml.Data;
+
+namespace DevHome.Common.Helpers;
+
+public class EnumToIntConverter : IValueConverter
```
Comment (dhoehna) [inline common/Helpers/EnumToIntConverter.cs:10]: Is this class just for converting an into to an enum and an enum to an int?  Why?  Does C# casting functionality lacking in a way?

File: common/Helpers/EnumToIntConverter.cs (hunk 1 above)
Comment (krschau) [reply common/Helpers/EnumToIntConverter.cs:12]: The converter I converted (ha) this from also had it.

File: common/Helpers/EnumToIntConverter.cs (hunk 2 above)
Comment (krschau) [reply common/Helpers/EnumToIntConverter.cs:10]: I couldn't get it to work with simple casting, but maybe I didn't do it the right way.

**Comment by krschau** [comment]
Converting to draft while additional design discussions happen.

File: common/Helpers/EnumToIntConverter.cs (hunk 2 above)
Comment (dhoehna) [reply common/Helpers/EnumToIntConverter.cs:10]: o.i.c.  This is a converted for the view.

File: settings/DevHome.Settings/Views/PreferencesPage.xaml [hunk 3]
```diff
@@ -26,44 +27,23 @@
...
-                                </RadioButton>
-                            </StackPanel>
-                        </labs:SettingsCard>
-                    </labs:SettingsExpander.Items>
-                </labs:SettingsExpander>
+                <labs:SettingsCard x:Uid="Settings_Theme">
+                    <labs:SettingsCard.HeaderIcon>
+                        <FontIcon Glyph="&#xE790;" />
+                    </labs:SettingsCard.HeaderIcon>
+                    <ComboBox x:Name="ThemeSelectionComboBox"
```
Comment (dhoehna) [inline settings/DevHome.Settings/Views/PreferencesPage.xaml:34]: NIT: I believe you can get around a custom converter by making a view model that houses the integer and string description and set the DataTemplate to the view model.

**Review by dhoehna — APPROVED** [review]

File: settings/DevHome.Settings/Views/PreferencesPage.xaml [hunk 4]
```diff
@@ -26,44 +27,23 @@
...
+                        <ComboBoxItem x:Uid="Settings_Theme_Default" />
+                        <ComboBoxItem x:Uid="Settings_Theme_Light" />
+                        <ComboBoxItem x:Uid="Settings_Theme_Dark" />
+                        <i:Interaction.Behaviors>
+                            <ic:EventTriggerBehavior EventName="SelectionChanged">
+                                <ic:InvokeCommandAction Command="{x:Bind ViewModel.SwitchThemeCommand}" 
+                                                        CommandParameter="{Binding SelectedIndex, ElementName=ThemeSelectionComboBox}" />
+                            </ic:EventTriggerBehavior>
+                        </i:Interaction.Behaviors>
+                    </ComboBox>
```
Comment (AmelBawa-msft) [inline settings/DevHome.Settings/Views/PreferencesPage.xaml:45]: Current approach works, but could be simplified as suggested below, allowing us to eliminate the dependency on a converter (which will be hard to maintain if we add a supported types list in it), and relax the constraint for listing the combobox items based on the values order.

```suggestion
                        <ComboBoxItem x:Uid="Settings_Theme_Default" Tag="{x:Bind ui:ElementTheme.Default}" />
                        <ComboBoxItem x:Uid="Settings_Theme_Light" Tag="{x:Bind ui:ElementTheme.Light}" />
                        <ComboBoxItem x:Uid="Settings_Theme_Dark" Tag="{x:Bind ui:ElementTheme.Dark}" />
                        <i:Interaction.Behaviors>
                            <ic:EventTriggerBehavior EventName="SelectionChanged">
                                <ic:InvokeCommandAction Command="{x:Bind ViewModel.SwitchThemeCommand}" 
                                                        CommandParameter="{x:Bind ((ComboBoxItem)ThemeSelectionComboBox.SelectedItem).Tag, Mode=OneWay}" />
                            </ic:EventTriggerBehavior>
                        </i:Interaction.Behaviors>
                    </ComboBox>
```

Comment (krschau) [reply settings/DevHome.Settings/Views/PreferencesPage.xaml:45]: I'm not sure how to get out of using the converter here -- the suggested code works for changing theme, but without the converter we can't start the combobox out with the correct item selected. I was hoping something like 
```
                    <ComboBox x:Name="ThemeSelectionComboBox"
                              SelectedValuePath="Tag"
                              SelectedValue="{x:Bind Mode=OneTime, Path=ViewModel.ElementTheme}">
```
would work, but it doesn't. Do you know what I'm doing wrong?

Comment (AmelBawa-msft) [reply settings/DevHome.Settings/Views/PreferencesPage.xaml:45]: Gotcha, if the initial value is not configurable from WinUI XAML, we may be able to configure this from code-behind or alternatively we can store the enum values in the view-model as a property, along with an observable property for the selected index (using two way binding).

File: settings/DevHome.Settings/Views/PreferencesPage.xaml (hunk 3 above)
Comment (krschau) [reply settings/DevHome.Settings/Views/PreferencesPage.xaml]: I'm removing the converter and going with a code behind solution for selecting the current theme on page load.

**Review by EricJohnson327 — APPROVED** [review]

File: common/Helpers/EnumToIntConverter.cs (hunk 1 above)
Comment (krschau) [reply common/Helpers/EnumToIntConverter.cs]: File deleted.

**Pull request merged**